examples/simple/src/posts/PostEdit.tsx

```diff
@@ -1,253 +1,240 @@
-                        validate={[required(), number(), minValue(0)]}
-                    />
-                    <BooleanInput source="commentable" defaultValue />
-                    <TextInput disabled source="views" />
-                </FormTab>
-                <FormTab label="post.form.comments">
-                    <ReferenceManyField
-                        reference="comments"
-                        target="post_id"
-                        addLabel={false}
-                        fullWidth
-                    >
-                        <Datagrid>
-                            <DateField source="created_at" />
-                            <TextField source="author.name" />
-                            <TextField source="body" />
-                            <EditButton />
-                        </Datagrid>
-                    </ReferenceManyField>
-                </FormTab>
-            </TabbedForm>
-        </Edit>
-    );
+                        {...rest}
+                      />
+                    ) : null
+                  }
+                </FormDataConsumer>
+              </SimpleFormIterator>
+            </ArrayInput>
+          )}
+        </FormTab>
+        <FormTab label="post.form.body">
+          <RichTextInput
+            source="body"
+            label=""
+            validate={required()}
+            addLabel={false}
+          />
+        </FormTab>
+        <FormTab label="post.form.miscellaneous">
+          <TagReferenceInput reference="tags" source="tags" label="Tags" />
+          <ArrayInput source="backlinks">
+            <SimpleFormIterator>
+              <DateInput source="date" />
+              <TextInput source="url" validate={required()} />
+            </SimpleFormIterator>
+          </ArrayInput>
+          <DateInput source="published_at" options={{ locale: "pt" }} />
+          <SelectInput
+            create={
+              <CreateCategory
+                // Added on the component because we have to update the choices
+                // ourselves as we don't use a ReferenceInput
+                onAddChoice={(choice) => categories.push(choice)}
+              />
+            }
+            allowEmpty
+            resettable
+            source="category"
+            choices={categories}
+          />
+          <NumberInput
+            source="average_note"
+            validate={[required(), number(), minValue(0)]}
+          />
+          <BooleanInput source="commentable" defaultValue />
+          <TextInput disabled source="views" />
+          <ArrayInput source="pictures">
+            <SimpleFormIterator>
+              <TextInput source="url" initialValue="" />
+              <ArrayInput source="metas.authors">
+                <SimpleFormIterator>
+                  <TextInput source="name" initialValue="" />
+                </SimpleFormIterator>
+              </ArrayInput>
+            </SimpleFormIterator>
+          </ArrayInput>
```

Comment:
Sorry, I also added this part (this is in fact the most important part of the bug report)